src/py/mat3ra/made/tools/build/deformation/configuration.py

```diff
@@ -0,0 +1,24 @@
+from typing import Callable, List, Dict, Tuple
+
+from mat3ra.code.entity import InMemoryEntity
+from mat3ra.made.material import Material
+from pydantic import BaseModel
+
+from ...utils import DeformationFunctionBuilder
+
+
+class DeformationConfiguration(BaseModel, InMemoryEntity):
```

Comment:
DeformationConfiguration with deformation_function

SlabDeformationConfiguration inheriting from it and adding `slab`

Same for builders